activerecord/test/cases/transaction_callbacks_test.rb

```diff
@@ -200,21 +199,20 @@ def test_only_call_after_rollback_on_create_after_transaction_rollsback_for_new_
   end
 
   def test_call_after_rollback_when_commit_fails
-    @first.class.connection.singleton_class.send(:alias_method, :real_method_commit_db_transaction, :commit_db_transaction)
-    begin
-      @first.class.connection.singleton_class.class_eval do
-        def commit_db_transaction; raise "boom!"; end
+    @first.after_commit_block{|r| r.history << :after_commit}
+    @first.after_rollback_block{|r| r.history << :after_rollback}
+    
+    saved = @first.transaction {
+      tx = @first.class.connection.transaction_manager.current_transaction
+      def tx.commit
+        raise
       end
 
-      @first.after_commit_block{|r| r.history << :after_commit}
-      @first.after_rollback_block{|r| r.history << :after_rollback}
+      @first.save
+    } rescue nil
```

Comment:
Here is the same, my bad, haven't paid enough attention, just upgraded the old test keeping its `rescue nil; assert !saved` stuff. I'll fix it